lib/active_resource/base.rb

```diff
@@ -1698,13 +1758,22 @@ def split_options(options = {})
         self.class.__send__(:split_options, options)
       end
 
+      def attribute=(name, value)
+        super
+      rescue ActiveModel::MissingAttributeError
+        # TODO uses private API to assign to unknown attributes without a schema
+        @attributes[name] = ActiveModel::Attribute.from_user(name, value, ActiveModel::Type.default_value)
+
+        retry
+      end
```

Comment:
Since `ActiveResource::Base` supports writing to and reading from attributes that are not part of any `Schema` declaration, this change adds them to the Attribute Set as they are assigned.

As an alternative to this approach (which relies on private APIs), known and unknown attributes could be represented by different instance variables (ActiveModel-backed `@attributes` and a raw `@unknown_attributes` Hash instance, maybe).